### PR TITLE
Cleanup rust client

### PR DESCRIPTION
**Pull request by binyebarwe**

### CODE REVIEW

**Comment by skrrb** [comment]
you should probably add this to the CI

File: lib/client/Cargo.toml [hunk 1]
```diff
@@ -19,15 +19,14 @@ futures = "0.3.25"
 itertools = "0.10.3"
 jsonrpc-core = "18.0.0"
 jsonrpc-core-client = { version = "18.0.0", features = ["ws", "http", "tls"] }
-openbook-v2 = { path = "../../programs/openbook-v2", features = ["client"] }
+openbook-v2 = { path = "../../programs/openbook-v2", features = ["client", "enable-gpl"] }
```
Comment (skrrb) [inline lib/client/Cargo.toml:22]: this goes against the license, and i don't think it's necessary at all

File: lib/client/Cargo.toml [hunk 2]
```diff
@@ -19,15 +19,14 @@ futures = "0.3.25"
...
 pyth-sdk-solana = "0.1.0"
-serum_dex = { git = "https://github.com/openbook-dex/program.git", default-features=false,features = ["no-entrypoint", "program"] }
 shellexpand = "2.1.0"
 solana-account-decoder = "~1.14.9"
-solana-client = "~1.14.9"
-solana-rpc = "~1.14.9"
-solana-sdk = "~1.14.9"
-solana-address-lookup-table-program = "~1.14.9"
+solana-client = "1.14.17"
```
Comment (skrrb) [inline lib/client/Cargo.toml:26]: should this be the same version (1.14.16) as in the program?

Comment (binyebarwe) [reply lib/client/Cargo.toml:26]: I don't see why

Comment (skrrb) [reply lib/client/Cargo.toml:26]: cause they're part of the same workspace

Comment (binyebarwe) [reply lib/client/Cargo.toml:26]: The smart contract has the option to be inmutable. Let's set that as the option we'll go with openbook, then you are saying is that in 1 year we have to still use the client on the version 1.14.16. Sorry but this makes no sense. The way you interact with a blockchain (client) can be different as the data saved on it.

Comment (skrrb) [reply lib/client/Cargo.toml:26]: the problem is that cargo unifies the dependencies in the workspace, so you lose control over which specific ones your program is using

**Review by skrrb — APPROVED** [review]

**Pull request merged**